src/struct.js

```diff
@@ -1,72 +1,76 @@
-import each from 'lodash/each';
-import map from 'lodash/map';
-import isUndefined from 'lodash/isUndefined';
-import fromPairs from 'lodash/fromPairs';
 import { Reference } from './reference';
-import includeIoMixin from './io-mixin';
+import { XdrPrimitiveType } from './xdr-type';
+import { XdrWriterError } from './errors';
 
-export class Struct {
+export class Struct extends XdrPrimitiveType {
   constructor(attributes) {
+    super();
     this._attributes = attributes || {};
   }
 
-  static read(io) {
-    const fields = map(this._fields, (field) => {
-      const [name, type] = field;
-      const value = type.read(io);
-      return [name, value];
-    });
-
-    return new this(fromPairs(fields));
+  /**
+   * @inheritDoc
+   */
+  static read(reader) {
+    const attributes = {};
+    for (const [fieldName, type] of this._fields) {
+      attributes[fieldName] = type.read(reader);
+    }
+    return new this(attributes);
   }
 
-  static write(value, io) {
-    if (!(value instanceof this)) {
-      throw new Error(`XDR Write Error: ${value} is not a ${this.structName}`);
+  /**
+   * @inheritDoc
+   */
+  static write(value, writer) {
+    if (!(value instanceof this))
+      throw new XdrWriterError(`${value} is not a ${this.structName}`);
+
+    for (const [fieldName, type] of this._fields) {
+      const attribute = value._attributes[fieldName];
+      type.write(attribute, writer);
     }
-    each(this._fields, (field) => {
-      const [name, type] = field;
-      const attribute = value._attributes[name];
-      type.write(attribute, io);
-    });
   }
 
+  /**
+   * @inheritDoc
+   */
   static isValid(value) {
     return value instanceof this;
   }
 
   static create(context, name, fields) {
-    const ChildStruct = class extends Struct {};
+    const ChildStruct = class extends Struct {
+    };
 
     ChildStruct.structName = name;
 
     context.results[name] = ChildStruct;
 
-    ChildStruct._fields = fields.map(([fieldName, field]) => {
+    const mappedFields = new Array(fields.length);
+    for (let i = 0; i < fields.length; i++) {
+      const fieldDescriptor = fields[i];
+      const fieldName = fieldDescriptor[0];
+      let field = fieldDescriptor[1];
```

Comment:
Would deconstruction work here?

```suggestion
      let [ fieldName, field ] = fields[i];
```

Reply:
That's exactly how I coded it from the first attempt. However, the linter was against it – in the linter's opinion, the `fieldName` is never changed, so it should be defined as `const`. I'm 💯 in favor of the proposed change, but the linter should be suppressed for the line in this case.